pkg/build/builder/common.go

```diff
@@ -82,3 +91,141 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// randomBuildTag generates a random tag used for building images in such a way
+// that the built image can be referred to unambiguously even in the face of
+// concurrent builds with the same name in the same namespace.
+func randomBuildTag(namespace, name string) string {
+	repo := fmt.Sprintf("%s/%s", namespace, name)
+	randomTag := fmt.Sprintf("%08x", rand.Uint32())
+	maxRepoLen := reference.NameTotalLengthMax - len(randomTag) - 1
+	if len(repo) > maxRepoLen {
+		repo = fmt.Sprintf("%x", sha1.Sum([]byte(repo)))
+	}
+	return fmt.Sprintf("%s:%s", repo, randomTag)
+}
+
+// containerNamePrefix prefixes the name of containers launched by a build. We
+// cannot reuse the prefix "k8s" because we don't want the containers to be
+// managed by a kubelet.
+const containerNamePrefix = "openshift"
+
+// containerName creates names for Docker containers launched by a build. It is
+// meant to resemble Kubernetes' pkg/kubelet/dockertools.BuildDockerName.
```

Comment:
In the case you describe, we should be deleting and force terminating the old container - it's "dead" because the actor (the builder pod) has a new state (updated build) but the old container represents a previous state.  So the correct behavior I believe would be that we detect a container already exists, we force kill it, and if we can't kill it we return an error failing the build for the admin to go clean up.  We should definitely not allow it to exist at the same time.